src/enzyme_ad/jax/Passes/EnzymeHLOOpt.cpp

```diff
@@ -2209,6 +2209,57 @@ struct DynamicUpdateSliceConstProp final
   }
 };
 
+struct LogConstProp final : OpRewritePattern<mlir::stablehlo::LogOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(mlir::stablehlo::LogOp op,
+                                PatternRewriter &rewriter) const override {
+    // return if not constant
+    DenseElementsAttr inputAttr;
+    if (!matchPattern(op.getOperand(), m_Constant(&inputAttr)))
+      return failure();
+
+    // get the resultType
+    auto resultType = op.getType().cast<ShapedType>();
+
+    // Convert constant to tensor, compute log, then convert back to attribute
+    stablehlo::Tensor inputTen = mlir::stablehlo::constantOp(inputAttr);
```

Comment:
Look at some of the other examples, can you add a fast path if the attribute is a splat (so we don’t need to compute the same value for a huge constant of the same inputs)

Reply:
Kind of like here? https://github.com/EnzymeAD/Enzyme-JAX/blob/fd186b7f8ba82c41bf89a2e623d009746f13426f/src/enzyme_ad/jax/Passes/EnzymeHLOOpt.cpp#L2186

Reply:
more like this: https://github.com/EnzymeAD/Enzyme-JAX/blob/fd186b7f8ba82c41bf89a2e623d009746f13426f/src/enzyme_ad/jax/Passes/EnzymeHLOOpt.cpp#L3237

Reply:
otherwise we end up iterating through all inputs: https://github.com/openxla/stablehlo/blob/c06b1a10d66069505ccbab72b21f6fa0547c617d/stablehlo/reference/Ops.cpp#L1911

Reply:
Ah I see so replacement will still be the same, we are just playing around with the tensor type